ReText/singleapplication.py

```diff
@@ -0,0 +1,167 @@
+# This file is part of ReText
+# Copyright: 2012-2015 Dmitry Shachnev
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import functools
+import struct
+from PyQt5.QtCore import pyqtSignal
```

Comment:
Please group the imports, i.e.:

    from PyQt5.QtCore import pyqtSignal, QObject, QSharedMemory, QSystemSemaphore